### PR TITLE
Update vllm server to be openai compatible

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

Update vllm server to be OpenAI compatible

A lot of this was nabbed from https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/api_server.py

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

File: model-engine/model_engine_server/inference/vllm/vllm_server.py [hunk 1]
```diff
@@ -200,23 +253,63 @@ def extract_logprobs(logprobs: Dict[int, Logprob]) -> Dict[int, float]:
...
+
+    signal.signal(signal.SIGUSR1, debug)
 
     engine_args = AsyncEngineArgs.from_cli_args(args)
     engine = AsyncLLMEngine.from_engine_args(engine_args)
 
-    signal.signal(signal.SIGUSR1, debug)
+    event_loop: Optional[asyncio.AbstractEventLoop]
+    try:
+        event_loop = asyncio.get_running_loop()
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/inference/vllm/vllm_server.py:285]: when would there be event loop?

**Review by yunfeng-scale — APPROVED** [review]

File: model-engine/model_engine_server/inference/vllm/vllm_server.py (hunk 1 above)
Comment (dmchoiboi) [reply model-engine/model_engine_server/inference/vllm/vllm_server.py:285]: I guess for our use case, there wouldn't be one

File: model-engine/model_engine_server/inference/vllm/vllm_server.py [hunk 2]
```diff
@@ -1,33 +1,86 @@
...
+    return Response(status_code=200)
+
+
+@app.get("/v1/models")
+async def show_available_models():
+    models = await openai_serving_chat.show_available_models()
+    return JSONResponse(content=models.model_dump())
+
+
+@app.post("/v1/chat/completions")
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/inference/vllm/vllm_server.py:63]: how would you use these APIs though? do you have follow up PRs?

Comment (dmchoiboi) [reply model-engine/model_engine_server/inference/vllm/vllm_server.py:63]: yes, will include in follow-up prs; just trying to keep these PRs small

**Pull request merged**